tasks/rh-sign-cosign-image/tests/test-rh-sign-image-single-component.yaml

```diff
@@ -0,0 +1,104 @@
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      taskSpec:
+        workspaces:
+          - name: data
+        steps:
+          - name: setup-values
+            image: quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+              _TEST_MANIFEST_DIGEST="sha256:2222"
+              _TEST_MANIFEST_REFERENCE="quay.io/test-product----test-image0@${_TEST_MANIFEST_DIGEST}"
+              _TEST_MANIFEST_REPO="registry.redhat.io/test-product/test-image0"
+              cat > $(workspaces.data.path)/snapshot_spec.json << EOF
+              {
+                "application": "myapp",
+                "components": [
+                  {
+                    "name": "comp0",
+                    "containerImage": "$_TEST_MANIFEST_REFERENCE",
+                    "repository": "$_TEST_MANIFEST_REPO"
+                  }
+                ]
+              }
+              EOF
+              cat > $(workspaces.data.path)/data.json << EOF
+              {
+                "sign": {
+                  "request": "hacbs-signing-pipeline",
+                  "configMapName": "signing-config-map"
+                }
+              }
+              EOF
+    - name: run-task
+      taskRef:
+        name: rh-sign-cosign-image
+      params:
+        - name: snapshotPath
+          value: snapshot_spec.json
+        - name: dataPath
+          value: data.json
+        - name: secretName
+          value: 'test-rh-sign-cosign-image-secret'
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      runAfter:
+        - setup
+    - name: check-result
+      workspaces:
+        - name: data
+          workspace: tests-workspace
+      taskSpec:
+        steps:
+          - name: check-result
+            image: quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+            script: |
+              #!/usr/bin/env sh
+              set -eux
+              echo "check results"
+              _TEST_PUB_REPO="registry.redhat.io/test-product/test-image0"
+              _TEST_REPO="quay.io/test-product----test-image0"
+
+              CALLS=$(cat "$(workspaces.data.path)/mock_skopeo_calls")
+              [ "$CALLS" = "inspect --raw docker://quay.io/test-product----test-image0@sha256:2222" ] || exit 1
+              CALLS=$(cat "$(workspaces.data.path)/mock_cosign_calls")
+              PRE_ARGS="AWS_REGION=us-test-1 AWS_ACCESS_KEY_ID=test-access-key AWS_SECRET_ACCESS_KEY=test-secret-access-key"
+              COMMON_ARGS="-t 3m0s sign --key aws://arn:mykey"
+              EXPECTED=$(cat <<EOF
+              $PRE_ARGS cosign $COMMON_ARGS --sign-container-identity ${_TEST_PUB_REPO}@sha256:2222 ${_TEST_REPO}@sha256:2222
+              EOF
+              )
+              [ "$CALLS" = "$EXPECTED" ] || exit 1
+      runAfter:
+        - run-task
+  finally:
```

Comment:
Don't think you need from here down (same comment as other test)

Reply:
When I run tests locally it keeps workspace for both test cases and therefore mock calls files contain records from previous test

Reply:
Are you sure? I don't know how that could be the case, since, as Johnny said, each pipeline run is tarted with its own PVC: https://github.com/konflux-ci/release-service-catalog/blob/development/.github/scripts/test_tekton_tasks.sh#L119
```sh
    PIPELINERUN=$(tkn p start $TEST_NAME -w name=tests-workspace,volumeClaimTemplateFile=$WORKSPACE_TEMPLATE -o json | jq -r '.metadata.name')
```

The template is here: https://github.com/konflux-ci/release-service-catalog/blob/development/.github/resources/workspace-template.yaml